src/app/components/AdvancedSearch/AdvancedSearch.jsx

```diff
@@ -225,13 +241,13 @@ class AdvancedSearch extends React.Component {
         }
       });
     }
-    // If viaf or lcnaf is set, but author is not set, they were publisher searches
-    // Since we don't give users a way to clear this, we should
-    // pre-emptively clear it.
-    if (!state.queries.author) {
-      state.queries.viaf = '';
-      state.queries.lcnaf = '';
-    }
+
+    const displaySearchQueries = searchQuery.queries.filter(query => searchFields.includes(query.field));
```

Comment:
Won't this preserve `lcnaf` and `viaf` fields in the `displayQueries` object? Or am I missing something?

Reply:
No `filter` checks for whether the field exists in `searchFields` ( https://github.com/NYPL/sfr-bookfinder-front-end/pull/150/files#diff-0565b36c278253c00836552fe5428f75R9 )

Reply:
ah ok! I got `searchFields` mixed up with `fields`